src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/ParserMin.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.RegularExpressions
+{
+    internal ref struct ParserMin
+    {
+        private readonly RegexOptions _options;
+        private InputWalker _pattern;
+
+        public int Position => _pattern.Position;
+
+        public ParserMin(ReadOnlySpan<char> pattern, RegexOptions options = RegexOptions.None)
+        {
+            _options = options;
+            _pattern = new InputWalker(pattern);
+        }
+
+        /// <summary>
+        /// Scans \ code for escape codes that map to single Unicode chars.
+        /// </summary>
+        public char ScanCharEscape()
+        {
+            char ch = _pattern.RightCharMoveRight();
+
+            if (ch >= '0' && ch <= '7')
```

Comment:
Use something like this https://github.com/dotnet/coreclr/blob/faa4c87ff3e025c9bed978522f3fe4c85ceecb60/src/System.Private.CoreLib/shared/System/Text/UTF8Encoding.cs#L836?

```C#
if ((uint)(ch - '0') <= (uint)('7' - '0'))
```